runtime/src/main/java/io/quarkiverse/semantickernel/semanticfunction/SemanticFunctionProducer.java

```diff
@@ -0,0 +1,131 @@
+import com.microsoft.semantickernel.semanticfunctions.SemanticFunctionConfig;
+import com.microsoft.semantickernel.skilldefinition.FunctionNotFound;
+import com.microsoft.semantickernel.textcompletion.CompletionRequestSettings;
+import com.microsoft.semantickernel.textcompletion.CompletionSKFunction;
+
+import io.quarkiverse.semantickernel.SemanticKernelConfiguration;
+import io.quarkiverse.semantickernel.semanticfunction.SemanticFunctionConfiguration.Skill.Function;
+
+public class SemanticFunctionProducer {
+
+    @Inject
+    Kernel kernel;
+    @Inject
+    SemanticKernelConfiguration configuration;
+
+    @Produces
+    @SemanticFunction
+    public CompletionSKFunction buildFunction(InjectionPoint ip) {
+        var skillName = getSkillName(ip);
+        var functionName = getFunctionName(ip);
+
+        if (isKnownSemanticFunction(skillName, functionName)) {
+            return kernel.getSkill(skillName).getFunction(functionName, CompletionSKFunction.class);
+        } else {
+            Optional<CompletionSKFunction> rv = loadFromDirectory(skillName, functionName);
+            if (rv.isPresent()) {
+                return rv.get();
+            } else {
+                return loadFromConfigurations(skillName, functionName);
+            }
+        }
+    }
+
+    private String getSkillName(InjectionPoint ip) {
+        var rv = ip.getAnnotated().getAnnotation(SemanticFunction.class).skill();
+        return rv != null && !rv.isBlank() ? rv : "default";
+    }
+
+    private String getFunctionName(InjectionPoint ip) {
+        var rv = ip.getAnnotated().getAnnotation(SemanticFunction.class).function();
+        return rv != null && !rv.isBlank() ? rv : "default";
+    }
+
+    private Optional<CompletionSKFunction> loadFromDirectory(String skillName, String functionName) {
+        var directory = configuration.semanticFunction().fromDirectory();
+        if (directory.isPresent()) {
+            kernel.importSkillsFromDirectory(directory.get(), skillName);
+            var knownFunction = isKnownSemanticFunction(skillName, functionName);
+            return knownFunction
+                    ? Optional.of(kernel.getSkill(skillName).getFunction(functionName, CompletionSKFunction.class))
+                    : Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    private boolean isKnownSemanticFunction(String skillName, String functionName) {
+        Predicate<SKFunction<?>> knownCompletionPredicate = f -> {
+            return f.getName().equals(functionName)
+                    && CompletionRequestSettings.class.equals(f.getType());
+        };
+        return kernel.getSkills().asMap().containsKey(skillName)
+                && kernel.getSkill(skillName).getAll().stream().anyMatch(knownCompletionPredicate);
+    }
+
+    private CompletionSKFunction loadFromConfigurations(String skillName, String functionName) {
+        var skillConfiguration = configuration.semanticFunction().skills().get(skillName);
```

Comment:
I'm in two minds with the usage of `var`. Semantic Kernel is supposed to work with Java 8, `var` was introduced in Java 10... but Quarkus 3 does only support 11+

So you will not find any `var` in Semantic Kernel APIs (just on the samples).

Reply:
I struggle to leave java 8 :) anyway, var removed